api/decorators.py

```diff
@@ -2,13 +2,15 @@
 from flask import abort
 from apifairy import arguments, response
 import sqlalchemy as sqla
+from sqlalchemy import or_
+
 from api.app import db
 from api.schemas import StringPaginationSchema, PaginatedCollection
 
 
 def paginated_response(schema, max_limit=25, order_by=None,
                        order_direction='asc',
-                       pagination_schema=StringPaginationSchema):
+                       pagination_schema=StringPaginationSchema, filter_fields=[]):
```

Comment:
This solution isn't very flexible, as it only allows for one search option per endpoint. If you also wanted to allow the client to also search users that registered between two dates, then you would be forced to create a separate endpoint just for that. Or if you wanted to let the client search by username only, and by email only, then those would be two additional endpoints. It doesn't scale well, and besides, searching shouldn't even need separate endpoints, that could be handled with query string arguments on the same endpoint.